File: OsmAnd/src/net/osmand/plus/track/data/TracksGroup.java
package net.osmand.plus.track.data;

import android.content.Context;

import androidx.annotation.NonNull;

import net.osmand.shared.gpx.TrackItem;

import java.util.List;

public interface TracksGroup {

	@NonNull
	String getName(@NonNull Context context);

	@NonNull
	List<TrackItem> getTrackItems();
}

File: OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/gpx/data/TracksGroup.kt
package net.osmand.shared.gpx.data

import net.osmand.shared.gpx.TrackItem
import net.osmand.shared.gpx.enums.TracksSortScope

interface TracksGroup {
	fun getId(): String

	fun getName(): String

	fun getTrackItems(): List<TrackItem>

	fun getSubgroupById(subgroupId: String): TracksGroup? = null

	fun getTracksSortScope() = TracksSortScope.TRACKS

	fun getSupportedSortScopes(): List<TracksSortScope> = listOf(TracksSortScope.TRACKS)
}
